server/consumer.go

```diff
@@ -307,8 +308,18 @@ type consumer struct {
 	// Ack queue
 	ackMsgs *ipQueue
 
-	// For stream signaling.
-	sigSub *subscription
+	// for stream signaling when multiple filters are set.
+	sigSubs []*subscription
+}
+
+// A single subject filter.
+type subjectFilter struct {
+	subject     string
+	hasWildcard bool
```

Comment:
This needed? Can we just calculate if not in hot path?

Regardless move to end so better packing.

Reply:
moved to the end.

It's calculated when creating consumer, so it can be used cheaply when fetching messages.
That part didn't change from single-consumer approach - it just was a singular `wc` field on `consumer`, for the only possible filter subject.

Reply:
Thanks.